DDRec/src/DetectorSurfaces.cpp

```diff
@@ -35,8 +35,10 @@ namespace dd4hep {
       const VolSurfaceList* vsL = volSurfaceList(det) ;
 
       try {
-        _sL = det.extension< SurfaceList >() ;
-
+        _sL = det.extension< SurfaceList >(false) ;
+        if (not _sL) {
+          _sL = det.addExtension<SurfaceList >(  new SurfaceList( true )  ) ;
+        }
```

Comment:
I still see error printouts, even with these changes. I haven't seen any runtime issues yet with ILD reco, I have to still check results.

Reply:
Which k4geo test are you running?

Reply:
I am running the ILD standard reco, where this also pops up after #1226

Reply:
It might call a different `extension<>()`
Can you pinpoint via gdb?

Reply:
In ILD reconstruction the issue is this extension here: https://github.com/iLCSoft/DDKalTest/blob/81e946b08c3d244c762714e707b640dae0b8152d/src/DDPlanarMeasLayer.cc#L110

That in turn is defined here:
https://github.com/AIDASoft/DD4hep/blob/b6f660a95bcd6270083d34443cd2fbbca4006e97/DDRec/include/DDRec/DetectorData.h#L505

I couldn't immediately find a similar hook as for the SurfaceList, but someone with a bit more knowledge about the internals probably has more success.

Reply:
```diff
- auto* ext = detElement.extension<dd4hep::rec::DoubleParameters>();
+ auto* ext = detElement.extension<dd4hep::rec::DoubleParameters>(false);
```
and then `if(ext)` instead of `catch`?

Reply:
If I have a `false` in there, what happens with previous versions of DD4hep? IIUC, I still get an exception, right? So, I would only have to add an additional nullptr check to have it silent for newer versions?!

Reply:
It still works in previous versions of DD4hep with the `false` and nullptr check. The `if (!alert) return 0;` was already present before.

Reply:
PS: for ten years
![image](https://github.com/AIDASoft/DD4hep/assets/2115797/43467d36-c3d0-46ac-80c5-24f2f8424707)

Reply:
Just for completeness: https://github.com/iLCSoft/DDKalTest/pull/16 fixes things for ILD reco